contracts/precompiled/CallSolanaCaller.sol

```diff
@@ -28,9 +34,48 @@ contract CallSolanaCaller {
     }
 
     function execute(uint64 lamports, bytes calldata instruction) public {
-        bytes32 returnData = bytes32(_callSolana.execute(lamports, instruction));
-        emit LogBytes(returnData);
+       bytes32 returnData = bytes32(_callSolana.execute(lamports, instruction));
+       emit LogBytes(returnData);
+    }
+
+    function executeInIterativeMode(uint256 iterations, uint64 lamports, bytes calldata instruction) public {
+        doIterativeActions(iterations);
+        execute(lamports, instruction);
+    }
+
+    function executeAndDoSomeIterativeActions(uint256 iterations, uint64 lamports, bytes calldata instruction) public {
+        execute(lamports, instruction);
+        doIterativeActions(iterations);
+    }
+
+    function executeMultipleCallsAndDoIterativeActions(uint256 calls, uint256 iterations, uint64 lamports, bytes calldata instruction) public {
```

Comment:
can we rename instruction to something else? because instructions is a term from solana and it is a bit confusing
this instruction count != real iterations for the transaction

Reply:
done